File: examportalserver/src/main/java/com/exam/examspringsever/services/QuizService.java
package com.exam.examspringsever.services;

import java.util.List;
import java.util.Set;

import com.exam.examspringsever.models.exam.Category;
import com.exam.examspringsever.models.exam.Quiz;

public interface QuizService {
	public Quiz addQuiz(Quiz quiz);
	public Quiz updateQuiz(Quiz quiz);
	public Set<Quiz> getQuizzes();
	public Quiz getQuiz(Long quizId);
	public void deleteQuiz(Long quizId);
	public List<Quiz> getQuizzesOfCategory(Category category);
	
	public List<Quiz> getActiveQuizzes();
	public List<Quiz> getActiveQuizzesOfCategory(Category c);
}

File: exam-portal-front-angular/src/app/MyServices/quiz.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { environment } from 'src/environments/environment';

@Injectable({
  providedIn: 'root'
})
export class QuizService {
  SERVER_URL =  environment.SERVER_URL
  constructor(private http:HttpClient) { }

  //Load all quizzes
  public loadQuizzes=()=>{
    return this.http.get(`${this.SERVER_URL}/quiz/`);
  }

  // Add Quizes
  public addQuiz=(quiz)=>{
    return this.http.post(`${this.SERVER_URL}/quiz/`, quiz);
  }
  //Delete Quizz
  public deleteQuiz=(quizId)=>{
    return this.http.delete(`${this.SERVER_URL}/quiz/${quizId}`);
  }

  //Get Single Quiz
  public getQuiz=(quizId)=>{
    return this.http.get(`${this.SERVER_URL}/quiz/${quizId}`);
  }

  //Update Quiz
  public updateQuiz=(quiz)=>{
    return this.http.put(`${this.SERVER_URL}/quiz/`,quiz);
  }

  //Quiz By category ID
  public quizByCategoryId=(categoryId)=>{
    return this.http.get(`${this.SERVER_URL}/quiz/category/${categoryId}`)
  }

  //Get Actve Quizzes
  public getActiveQuizzes=()=>{
    return this.http.get(`${this.SERVER_URL}/quiz/active`);
  }

  //Get Active Quizzes of Category
  public getActiveQizzesOfCategory=(categoryId)=>{
    return this.http.get(`${this.SERVER_URL}/quiz/active/${categoryId}`);
  }
}
